docs/helpers.rst

```diff
@@ -191,6 +191,26 @@ transaction support.  This is only required for fixtures which need
 database access themselves.  A test function would normally use the
 :py:func:`~pytest.mark.django_db` mark to signal it needs the database.
 
+``shared_db_wrapper``
+~~~~~~~~~~~~~~~~~~~~~
+
+This fixture can be used to create long-lived state in the database.
+It's meant to be used from fixtures with scope bigger than ``function``.
+It provides a context manager that will create a new database savepoint for you,
+and will take care to revert it when your fixture gets cleaned up.
+
+At the moment it does not work with ``transactional_db``,
+as the fixture itself depends on transactions.
+It also needs Django >= 1.6, as it's based on the ``atomic`` API.
+
+Example usage::
+
+  @pytest.fixture(scope='module')
+  def some_users(request, shared_db_wrapper):
+      with shared_db_wrapper(request):
```

Comment:
I have to admit that the interactions between fixtures of different scopes are totally over my head :-/

The end goal is to call `transaction.atomic` and `transaction.set_rollback` as follows -- but I don't know how to map this to pytest's hooks and to make it work within pytest-django's existing infrastructure.

```
with transaction.atomic():
    fixture_1()     # used in the entire test suite

    with transaction.atomic():
        fixture_2()     # used in one module

        with transaction.atomic():
            test_21()
            transaction.set_rollback(True)

        with transaction.atomic():
            test_22()
            transaction.set_rollback(True)

        with transaction.atomic():
            test_23()
            transaction.set_rollback(True)

        transaction.set_rollback(True)  # fixture 2

    with transaction.atomic():
        fixture_3()     # used in two modules

        with transaction.atomic():
            fixture_4()     # used in one module which also uses 3

            with transaction.atomic():
                test_41()
                transaction.set_rollback(True)

            with transaction.atomic():
                test_42()
                transaction.set_rollback(True)

            with transaction.atomic():
                test_43()
                transaction.set_rollback(True)

            transaction.set_rollback(True)  # fixture 4

        with transaction.atomic():
            fixture_5()     # used in one module which also uses 3

            with transaction.atomic():
                test_51()
                transaction.set_rollback(True)

            with transaction.atomic():
                test_52()
                transaction.set_rollback(True)

            with transaction.atomic():
                test_53()
                transaction.set_rollback(True)

                transaction.set_rollback(True)

            transaction.set_rollback(True)  # fixture 5

        transaction.set_rollback(True)  # fixture 3

    transaction.set_rollback(True)  # fixture 1

```

If multiple databases are being used, you need to call `transaction.atomic(using=...)` and `transaction.set_rollback(True, using=...)` for each of them.

Ping me if you have questions about Django's transaction management — I wrote it.